proxywasm/proxytest/wasmwrapper.go

```diff
@@ -0,0 +1,423 @@
+// Copyright 2020-2022 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxytest
+
+import (
+	"context"
+	"reflect"
+	"unsafe"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/wasi_snapshot_preview1"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/internal"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+type guestABI struct {
+	proxyOnVMStart          api.Function
+	proxyOnContextCreate    api.Function
+	proxyOnConfigure        api.Function
+	proxyOnDone             api.Function
+	proxyOnQueueReady       api.Function
+	proxyOnTick             api.Function
+	proxyOnRequestHeaders   api.Function
+	proxyOnRequestBody      api.Function
+	proxyOnRequestTrailers  api.Function
+	proxyOnResponseHeaders  api.Function
+	proxyOnResponseBody     api.Function
+	proxyOnResponseTrailers api.Function
+	proxyOnLog              api.Function
+}
+
+type vmContext struct {
+	runtime wazero.Runtime
+	abi     guestABI
+}
+
+// NewWasmVMContext returns a types.VMContext that delegates plugin invocations to the provided compiled wasm binary.
+// proxytest can be run with a compiled wasm binary by passing this to proxytest.WithVMContext.
+// Note: Currently only HTTP plugins are supported.
+func NewWasmVMContext(wasm []byte) (types.VMContext, error) {
```

Comment:
I think we should retrun closure of runtime 
```suggestion
func NewWasmVMContext(wasm []byte) (types.VMContext, func() error, error) {
```

Reply:
Sure, went ahead and defined a type embedding `VMContext` and `Closer`